torch/optim/optimizer.py

```diff
@@ -188,8 +227,8 @@ def __init__(self, params, defaults):
                             "an iterable of Tensors or dicts, but got " +
                             torch.typename(params))
 
-        self.state = defaultdict(dict)
-        self.param_groups = []
+        self.state: DefaultDict[str, Dict[Any, Any]] = defaultdict(dict)
```

Comment:
Wait a moment...is the key str here? I believe it is usually a parameter (tensor) when the state gets initialized in _init_group() of the optimizers (see adamw.py for an example).

Reply:
oops you're right these definitely look like Tensor keys

Reply:
It gets real confusing because by the time it goes to a state_dict,  the representation is usually an int, and when other people superclass this, it can be a fully qualified name (or str).